packages/security/src/CredentialManagerOverride.ts

```diff
@@ -0,0 +1,254 @@
+    }
+
+    //________________________________________________________________________
+    /**
+     * Get the credential manager information for the specified plugin.
+     *
+     * @param pluginName - Name of the plugin package
+     *
+     * @returns An ICredentialManagerNameMap or
+     *          null if the specified plugin is not a known credential manager.
+     */
+    public static getCredMgrInfoByPlugin(pluginName: string) : ICredentialManagerNameMap | null {
+        let credMgrInfo: ICredentialManagerNameMap = null;
+        for (const nextNameMap of this.KNOWN_CRED_MGRS) {
+            if (nextNameMap.credMgrPluginName === pluginName) {
+                credMgrInfo = nextNameMap;
+                break;
+            }
+        }
+        return credMgrInfo;
+    }
+
+    //________________________________________________________________________
+    /**
+     * Get the credential manager information for the specified plugin.
+     *
+     * @param ZEExtName - Name of the Zowe Explorer extension
+     *
+     * @returns An ICredentialManagerNameMap or
+     *          null if the specified extension is not a known credential manager.
+     */
+    public static getCredMgrInfoByZEExt(ZEExtName: string) : ICredentialManagerNameMap | null {
+        let credMgrInfo: ICredentialManagerNameMap = null;
+        for (const nextNameMap of this.KNOWN_CRED_MGRS) {
+            if (nextNameMap.credMgrZEName === ZEExtName) {
+                credMgrInfo = nextNameMap;
+                break;
+            }
+        }
+        return credMgrInfo;
+    }
+
+    //________________________________________________________________________
+    /**
+     * Get the known credential managers.
+     *
+     * @returns An array of credential managers.
+     */
+    public static getKnownCredMgrs() : ICredentialManagerNameMap[] {
+        return this.KNOWN_CRED_MGRS;
+    }
+
+    /**
+     * Override the current credential manager with the specified
+     * credential manager. A plugin or ZE extension that provides a
+     * credential manager would override the default credential manager
+     * upon installation.
+     *
+     * @param newCredMgrName
+     *        The display name of your credential manager.
+     *
+     * @throws An ImperativeError upon error.
+     */
+    public static overrideCredMgr(newCredMgrName: string) : void {
+        const credMgrInfo: ICredentialManagerNameMap =
+            CredentialManagerOverride.getCredMgrInfoByDisplayName(newCredMgrName);
+        if (credMgrInfo === null) {
+            // We do not have a known credMgr. Form a list of known credential managers.
+            const knownCredMgrs: ICredentialManagerNameMap[] = CredentialManagerOverride.getKnownCredMgrs();
+            let knownMgrsMsg: string = "";
+            for ( let credMgrInx = 0; credMgrInx < knownCredMgrs.length; credMgrInx++) {
+                knownMgrsMsg += `\n${knownCredMgrs[credMgrInx].credMgrDisplayName}`;
+            }
```

Comment:
can be simplified with:

```TS
const knownMgrsMsg = knownCredMgrs.map(knowCredMgr => `\n${knownCredMgr.credMgrDisplayName}`).join('');
```

or

```TS
let knownMgrsMsg = '';
knownCredMgrs.forEach(knownCredMgr => knownMgrsMsg += `\n${knownCredMgr.credMgrDisplayName}`);
```

Reply:
here's another option if interested, but they are all functionally equivalent 😅 

```ts
knownCredMgrs.reduce(str, knowCredMgr => str.concat("\n", knownCredMgr.credMgrDisplayName), "")
```

Reply:
I made the suggested changes.